internal/featuredetection/feature_detection_test.go

```diff
@@ -348,3 +350,54 @@ func TestRepositoryFeatures(t *testing.T) {
 		})
 	}
 }
+
+func TestProjectV1SupportEnvOverride(t *testing.T) {
+	// Given PROJECT_V1_SUPPORTED env is set at all
+	t.Setenv("PROJECT_V1_SUPPORTED", "anyval")
```

Comment:
Note that I made a minor adjustment here because I didn't think it really mattered whether it was a boolean and it just added an extra error path. Don't feel strongly though.